hw/ip/power_manager/data/power_manager.hjson.tpl

```diff
@@ -60,6 +60,28 @@
     }
 
 % endfor
+% for bank in range(ram_numbanks):
+    { name:     "SET_RETENTIVE_RAM_BLOCK_${bank}",
+      desc:     "Used to set in retentive mode ram block ${bank}",
+      resval:   "0x00000000"
+      swaccess: "rw",
+      hwaccess: "hro",
+      fields: [
+        { bits: "0", name: "SET_RETENTIVE_RAM_BLOCK_${bank}", desc: "Set Retentive Ram Block ${bank} Reg" }
+      ]
+    }
+
+% endfor
+    { name:     "POWER_GATE_EXTERNAL",
+      desc:     "Used to power gate external domain",
+      resval:   "0x00000000"
+      swaccess: "rw",
+      hwaccess: "hro",
+      fields: [
+        { bits: "0", name: "POWER_GATE_EXTERNAL", desc: "Power Gate External Reg" }
```

Comment:
here I would put for example 31:0 and we can control at most 32 external power domains.
You can thus set the ext domain signals to 31:0, without parameter for now

Reply:
Mmm, yes, but untile now we always dedicated a single register per domain. I think it is easier to implement and manage by sw.